src/main.rs

```diff
@@ -58,21 +59,55 @@ impl LanguageServer for Backend {
             .to_file_path()
             .unwrap()
             .with_extension("pdf");
-
+        let messages: Vec<_>;
         let output = match typst::compile(world) {
```

Comment:
The new `messages` system should replace the old `output`, which was just a hack to get things working quickly. Remove the`output` variable and all references to it, and have the `match` expression evaluate to `messages` instead of setting it inside each branch.